### PR TITLE
MESG application in Aragon on testnet

### PR DESCRIPTION
**Pull request by blackinno**
- Using the Aragon application version on Rinkeby.
- Publish MESG on the Aragon IPFS network.

### CODE REVIEW

File: aragon-application/app/src/utils/processTemplate.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { process } from '@mesg/compiler'
+import { save } from './ipfs-util'
+
+const compiler = async ({ appAddress, name, mesgAddress, url, connectedAccount }, eventAbi, eventSignature) => {
+  const ethereumHash = '6d4MZPQR9MTcCN8sHAu2Vqb4yuWki7t2ktnFALt7x6Xg'
```
Comment (antho1404) [inline aragon-application/app/src/utils/processTemplate.js]: Can we put that in the config somewhere?
This way if we change the hash of the instance we can just change it in the config and not find it in the code.

Comment (blackinno) [reply aragon-application/app/src/utils/processTemplate.js]: I have tried many ways to add environment variables like `.env` file, using `REACT_APP_HASH` like `react-create-app`[document](https://create-react-app.dev/docs/adding-custom-environment-variables/#docsNav) said or add environment variables on the script like `REACT_APP_HASH=dhalshfwkjds parcel build index.html -d ../dist/ --public-url \".\" --no-cache` when building the front-end environment variables aren't assigned.

File: aragon-application/app/src/utils/processTemplate.js [hunk 2]
```diff
@@ -2,20 +2,24 @@ import { process } from '@mesg/compiler'
...
 
 const compiler = async ({ appAddress, name, mesgAddress, url, connectedAccount }, eventAbi, eventSignature) => {
-  const ethereumHash = '6d4MZPQR9MTcCN8sHAu2Vqb4yuWki7t2ktnFALt7x6Xg'
-  const webhookHash = 'E2oGXehaZqrX1fSPTSMDzMP7D6GmBpVNr2wZCyj8cfXE'
   const template = `
 name: ${name}-${mesgAddress}-${eventAbi.name}-${connectedAccount}
 steps:
 - type: trigger
-  instanceHash: ${ethereumHash}
+  instance:
```
Comment (antho1404) [inline aragon-application/app/src/utils/processTemplate.js]: This one should stay with `instanceHash` as it requires an already running service.
Could you revert this one and add the 2 hashes (ethereumHash and webhookHash) as environment like the IPFS path

File: aragon-application/README.md [hunk 3]
```diff
@@ -2,88 +2,31 @@
...
-      transactionHash: {key: transactionHash}
-      transactionIndex: {key: transactionIndex}
-      blockHash: {key: blockHash}
-      blockNumber: {key: blockNumber}
-  `
-```
+### Create configuration file
 
-[IPFS](app/src/utils/ipfs-util.js) File:
+Copy the `.env.example` to `.env` in folder `app`
```
Comment (antho1404) [inline aragon-application/README.md]: There is no `.env.example` in this directory

File: aragon-application/README.md [hunk 4]
```diff
@@ -2,88 +2,31 @@
...
 ```sh
-$ npm start
-```
-
-**_or_**
-
-```sh
-npm start:ipfs:template
+$ npm install
+$ npm start or npm start:ipfs:template
```
Comment (antho1404) [inline aragon-application/README.md]: ### `npm start` I have the following error

```
➜  aragon-application git:(aragon-testnet-abi) ✗ npm start              

> mesg@1.0.0 start /Users/antho/prog/aragon/aragon-application
> npm run start:ipfs


> mesg@1.0.0 start:ipfs /Users/antho/prog/aragon/aragon-application
> aragon run --files dist

Error: Cannot find module 'ganache-core'
Require stack:
- /Users/antho/prog/aragon/aragon-application/node_modules/@aragon/cli/dist/commands/devchain_cmds/start.js
- /Users/antho/prog/aragon/aragon-application/node_modules/@aragon/cli/dist/commands/devchain.js
```

### `npm start:ipfs:template`
not a valid command I guess what you want is `npm run start:ipfs:template`

File: aragon-application/README.md [hunk 5]
```diff
@@ -92,7 +35,13 @@ This MESG Aragon app is published in the AragonPM package manager on Rinkeby, so
...
+### Install Aragon cli
+
+```sh
+npm install -g @aragon/cli
+```
+
+### Installed application
 
 ```sh
 $ npx dao install <dao-name>.aragonid.eth mesg.open.aragonpm.eth --environment aragon:rinkeby
```
Comment (antho1404) [inline aragon-application/README.md]: Error running this command
```
➜  aragon-application git:(aragon-testnet-abi) ✗ npx dao install testmesgprocess.aragonid.eth mesg.open.aragonpm.eth --environment aragon:rinkeby
Error: Cannot find module 'ganache-core'
Require stack:
- /Users/antho/prog/aragon/aragon-application/node_modules/@aragon/cli/dist/commands/devchain_cmds/start.js
```

File: aragon-application/app/src/utils/processTemplate.js (hunk 2 above)
Comment (blackinno) [reply aragon-application/app/src/utils/processTemplate.js]: Done.

File: aragon-application/README.md (hunk 4 above)
Comment (blackinno) [reply aragon-application/README.md]: Can't reproduce the error `Error: Cannot find module 'ganache-core'`

File: aragon-application/README.md (hunk 5 above)
Comment (blackinno) [reply aragon-application/README.md]: Can't reproduce the error `Error: Cannot find module 'ganache-core'`

**Pull request merged**